essential-projects/metadata_contracts
# add interfaces for namespace factory

## What did you change?

This adds a factory to allow for reusable namespace decorators.
The factory produces a wrapper that contains the namespace information and re-exposes the metadata decorators. This way the namespace information does not need to be repeated on every decorator usage.

## How can others test the changes?

Currently this cannot be fully tested, because it is blocked by https://github.com/essential-projects/data_model/issues/2

Also this blocks the PR for https://github.com/essential-projects/metadata/pull/1

But since this is an isolated functionality that doesn't change the usual behavior - I think it should be reviewable. The only test so far would be to debug into the application and check whether or not the metadata provider contains the correct data after the startup phase (when the static code has been executed)

## PR-Checklist

Please check the boxes in this list after submitting your PR:
